std/conv.d

```diff
@@ -1555,6 +1556,17 @@ if (!isImplicitlyConvertible!(S, T) &&
     }
 }
 
+/**
+Void array to const array of ubyte conversion. Converting to a mutable array
+is not possible because source may contain pointers.
+ */
+private T toImpl(T, S)(S value) @trusted
+if (isArray!S && is(Unqual!(ElementType!S) == void) &&
+    isArray!T && is(ElementType!T == const ubyte) || is(ElementType!T == immutable ubyte))
+{
+   return cast(const(ubyte)[]) value.dup;
```

Comment:
`.to!T` should not allocate behind the user's back. I suggest you drop `.dup` and allow only conversion to `const ubyte[]`, which is safe if the input is not `shared`.

Reply:
There was not real valid to dup anyway since the result is not modifiable.